### ISSUE TITLE
error for adding external Javascript

### ISSUE THREAD

**Issue by z267xu** [plotly/dashR#91]
1. If we set external Javascript in 
`Dash$new(external_scripts = "your_own_js")`,
we would have error like:

_"Error in as.character(function (description, open = "", blocking = TRUE,  : cannot coerce type 'closure' to vector of type 'character'"_

The error comes from `generate_js_dist_html` function in `dashR` package utils file. The reason is the typo in `sprintf`, which should be **href** instead of **url**.

https://github.com/plotly/dashR/blob/6bc44b5c4203b03bd5ef04ab5dd24b23bf9b987b/R/utils.R#L513-L520

2. Also if we mix resources with attributes like `src`, `integrity`, `crossorigin`, etc, like:

`external_scripts <- list(list(`
`src = 'https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.10/lodash.core.js', `
`integrity = 'sha256-Qqd/EfdABZUcAxjOkMi8eGEivtdTkh3b65xCZL4qAQA=', `
`crossorigin = 'anonymous'))`

`Dash$new(external_scripts = external_scripts)`

we may have error like

"warning: the condition has length > 1 and only the first element will be used from if (grepl("^(?:http(s)?:\\/\\/)?[\\w.-]+(?:\\.[\\w\\.-]+)+[\\w\\-\\._~:/?#[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$",
error: values must be length 1,
 but FUN(X[[2]]) result is length 4 from vapply(self$config$external_stylesheets, generate_css_dist_html,"

I guess we should manipulate input `href` before we call function `generate_js_dist_html()` and `generate_css_dist_html()`

https://github.com/plotly/dashR/blob/6bc44b5c4203b03bd5ef04ab5dd24b23bf9b987b/R/utils.R#L492-L532

@rpkyle

**Comment by rpkyle** [comment]
@z267xu Thanks for this report, very helpful.

(1) I had run into the `href` vs. `url` issue previously, and I think I resolved it locally but the commits hadn't been pushed yet. I'll open a PR and aim to merge the changes into the `dev` branch.

(2) These attributes were added to Dash for Python in August 2018, so I should've incorporated them in Dash for R as well! Thanks for noting this. I think it might merit an issue so I don't lose track of it, will try to fix in the next week. It'll get added to the same PR I use for (1).

https://github.com/plotly/dash/blob/558047fe831f2dfe1a063af317787388bf391c10/dash/dash.py#L185-L189
